File: leetcode/solutions/0677.MapSumPairs/MapSum_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import MapSum from "./MapSum.ts";

Deno.test("0677. Map Sum Pairs", () => {
  const mapSum = new MapSum();

  mapSum.insert("apple", 3);
  assertEquals(mapSum.sum("ap"), 3);

  mapSum.insert("app", 2);
  assertEquals(mapSum.sum("ap"), 5);
});

File: leetcode/solutions/0677.MapSumPairs/MapSum_test.go
package solution

import (
	"testing"
)

func Test_0677_MapSumPairs(t *testing.T) {
	t.Parallel()

	var got, want int

	mapSum := Constructor()
	mapSum.Insert("apple", 3)
	got, want = mapSum.Sum("ap"), 3

	if got != want {
		t.Errorf("got %#v want %#v", got, want)
	}

	mapSum.Insert("app", 2)
	got, want = mapSum.Sum("ap"), 5

	if got != want {
		t.Errorf("got %#v want %#v", got, want)
	}
}
